pygfunction/gfunction.py

```diff
@@ -2067,7 +2085,8 @@ def _find_axial_borehole_pairs(self, boreholes):
                     m = borehole_to_self[k][0]
                     # Add the borehole to the group if a similar borehole is
                     # found
-                    if self._compare_boreholes(boreholes[i], boreholes[m]):
+                    if (self._compare_boreholes(boreholes[i], boreholes[m]) and
+                        self.nBoreSegments[i] == self.nBoreSegments[m]):
```

Comment:
It appears that _find_axial_borehole_pairs was not modified, but rather an addition to the _compare_borehole statement.